docs: Custom cosmjs interfaces tutorial update (#38)
cosmos/developer-portal: hands-on-exercise/3-cosmjs-adv/1-cosmjs-objects.md

Create a folder named `scripts` in your project root. This is where you will launch the Protobuf compilation:

```sh
$ mkdir scripts
```

In the `scripts` folder, install telescope for Protobuf compilation:

```sh
$ cd scripts
$ npm install --save-dev --save-exact @cosmology/telescope@1.0.4

> $ cd ..

Reviewer: I think you can remove this one as you also do it in `DockerGroup`'s "Local".

$ curl https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.45.4/third_party/proto/gogoproto/gogo.proto -o proto/gogoproto/gogo.proto
```

Now we're ready to compile:

Create a Telescope config `.telescope.json` with following content:

> ```json

Author: Memo to update a link of config after PR in checkers-academy merged

$ curl https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.45.4/third_party/proto/gogoproto/gogo.proto -o proto/gogoproto/gogo.proto
```

Now we're ready to compile:

> Create a telescope config `.telescope.json` with following content:

Reviewer: I think it would be worth adding a note after this JSON pointing out important elements. I am thinking of `protoDirs` and `outPath` first. But perhaps you have further ideas.

}
```

<HighlightBox type="note">

> There're three main sections in the config file:
> 
> * protoDirs: root dirs that contains folders with proto files. e.g. If protoDirs're set with `proto` and `proto-common`, when importing from `cosmos/base/query/v1beta1/pagination.proto`, telescope will go checking whether there's `proto/cosmos/base/query/v1beta1/pagination.proto` or `proto-common/cosmos/base/query/v1beta1/pagination.proto`, and then generate dependency based on the file.
> 
> * outPath: it contains ts files structured by the folder structure of proto files.
> 
> * options: telescope options for users to decide which part of the code will be generated and the way of how code will be transpiled. For more details please see: [telescope options](https://github.com/cosmology-tech/telescope#options)

Reviewer: ```suggestion
There are three main sections in the config file:

* `protoDirs`: root directory that contains folders with Protobuf files. e.g. If `protoDirs` is set with `proto` and `proto-common`, when importing from `cosmos/base/query/v1beta1/pagination.proto`, telescope will go checking whether there's `proto/cosmos/base/query/v1beta1/pagination.proto` or `proto-common/cosmos/base/query/v1beta1/pagination.proto`, and then generate a dependency based on the file.
* `outPath`: it contains Typescript files structured by the folder structure of Protobuf files.
* `options`: telescope options for users to decide which part of the code will be generated and the way of how code will be transpiled. For more details please see: [telescope options](https://github.com/cosmology-tech/telescope#options).
```

Also, should we capitalise Telescope?

    -v $(pwd):/checkers \
    -w /checkers/scripts \
    checkers_i \
    npx telescope transpile --config .telescope.json
```

</CodeGroupItem>

</CodeGroup>

After it's been ran successfully, there'll be a message:

```txt
✨ files transpiled in '/client/src/types/generated'
✨ transpilation successful!
```

You should now have your TypeScript files.

In order to easily repeat these steps in the future, you can add them to your existing `Makefile` with slight modifications:

> ```lang-makefile [https://github.com/cosmos/b9-checkers-academy-draft/blob/cosmjs-elements/Makefile#L6-L31]

Author: Memo to update the link of makefile after PR in checkers-academy merged

    mkdir -p proto/gogoproto
    curl https://raw.githubusercontent.com/cosmos/cosmos-sdk/${cosmos-version}/third_party/proto/gogoproto/gogo.proto -o proto/gogoproto/gogo.proto

gen-proto-ts: download-cosmos-proto install-gen-tool

>     npx telescope transpile --config .telescope.json

Reviewer: Should you add a `cd scripts`?
